OpenRA.Mods.Common/Activities/Transform.cs

```diff
@@ -122,8 +122,10 @@ void DoTransform(Actor self)
 				var health = self.TraitOrDefault<Health>();
 				if (health != null)
 				{
-					var newHP = ForceHealthPercentage > 0 ? ForceHealthPercentage : (health.HP * 100) / health.MaxHP;
-					init.Add(new HealthInit(newHP));
+					// Cast to long to avoid overflow when multiplying by the health
+					var hp = (long)health.HP;
+					var newHP = ForceHealthPercentage > 0 ? ForceHealthPercentage : (hp * 100) / health.MaxHP;
```

Comment:
This can be simplified to
```csharp
var newHP = ForceHealthPercentage > 0 ? ForceHealthPercentage : (int)(health.HP * 100L / health.MaxHP);
```

and the cast below dropped.  Likewise in the other cases below.